File: java/simple_numbers.java

class simple_numbers {
  public static void main(String[] args)
  {
    final int ARGUMENTS_START_SINCE = 0;
    final int REQUIRED_ARGUMENTS_COUNT = 1;
    final int RETCODE__NOT_ENOUGHT_ARGUMENTS = 1;
    final int RETCODE__AMOUNT_IS_NOT_INTEGER = 2;

    if (args.length < ARGUMENTS_START_SINCE+REQUIRED_ARGUMENTS_COUNT) {
      System.out.println("ERROR! Not enought arguments.");
      help();
      System.exit(RETCODE__NOT_ENOUGHT_ARGUMENTS);
    }

    int targetNumbersCount = 0;
    try {
      targetNumbersCount = Integer.parseInt(args[ARGUMENTS_START_SINCE+0]);
    }
    catch (Exception e) {
      String msg = String.format("ERROR! Cannot parse \"%s\" as integer.", args[ARGUMENTS_START_SINCE+0]);
      System.out.println(msg);
      help();
      System.exit(RETCODE__AMOUNT_IS_NOT_INTEGER);
    }

    boolean needToPrint = false;
    if (args.length >= ARGUMENTS_START_SINCE+2) {
      if (args[ARGUMENTS_START_SINCE+1].equalsIgnoreCase("print")) {
        needToPrint = true;
      }
    }

    String msg;
    msg = String.format("Seek %s simple numbers...", targetNumbersCount);
    System.out.println(msg);

    int foundNumbersCount = 0;
    int someNumber;
    boolean simple;
    int[] NumbersList;
    long beginTime = System.currentTimeMillis();

    NumbersList = new int[targetNumbersCount];
    NumbersList[foundNumbersCount] = 2;
    someNumber = NumbersList[foundNumbersCount];
    foundNumbersCount++;
    while (foundNumbersCount < targetNumbersCount) {
      someNumber++;
      simple = isNumberSimple(someNumber, foundNumbersCount, NumbersList);
      if (simple) {
        //найдено очередное простое число - добавить его в массив простых чисел
        NumbersList[foundNumbersCount++] = someNumber;
      }
    }
    long completeTime = System.currentTimeMillis();
    long durationMillis = completeTime - beginTime;
    double durationSeconds = durationMillis / 1000.0;

    if (needToPrint)
      printResults(NumbersList);

    msg = "Duration: " +  Double.toString(durationSeconds) + " seconds.";
    msg += " " + "Last simple number = " + Integer.toString(NumbersList[foundNumbersCount-1]);
    System.out.println(msg);
    System.exit(0);
  }

  public static void help()
  {
        System.out.println("simple_numbers <count> [print]");
        System.out.println("  count   amount of simple numbers");
        System.out.println("  print   print found simple numbers to stdout");
  }

  public static void printResults(int[] Items)
  {
    String msg;
    int n = 0;
    for (Integer x:Items) {
        n++;
        msg = Integer.toString(n) + ")\t" + Integer.toString(x);
        System.out.println(msg);
    }
  }

  public static boolean isNumberSimple(int checkingNumber, int numbersCount, int[] NumbersList)
  {
    int remainder;
    for (int i=0; i<numbersCount; i++) {
      remainder = checkingNumber % NumbersList[i];
      if (remainder == 0)
        return false; //проверяемое число делится на одно из найденных ранее простых чисел без остатка - значит оно не является простым
    }
    //проверяемое число не делится без остатка ни на одно из найденных ранее простых чисел - это число являетя простым
    return true;
  }
}

File: python3/simple_numbers.py
#!/usr/bin/env python3

import sys
import time

ARGUMENTS_START_SINCE = 1
REQUIRED_ARGUMENTS_COUNT = 1
RETCODE__NOT_ENOUGHT_ARGUMENTS = 1
RETCODE__AMOUNT_IS_NOT_INTEGER = 2


def help():
    print("simple_numbers <count> [print]");
    print("  count   amount of simple numbers");
    print("  print   print found simple numbers to stdout");


def print_results(Items):
    for i in range(len(Items)):
        msg = str(i+1) + ")\t" + str(Items[i])
        print(msg);


def is_number_simple(checking_number, numbers_list):
    for x in numbers_list:
      remainder = checking_number % x
      if remainder == 0:
        return False; #проверяемое число делится на одно из найденных ранее простых чисел без остатка - значит оно не является простым
    #проверяемое число не делится без остатка ни на одно из найденных ранее простых чисел - это число являетя простым
    return True;


if __name__ == "__main__":
    if len(sys.argv) < ARGUMENTS_START_SINCE + REQUIRED_ARGUMENTS_COUNT:
        print("ERROR! Not enought arguments")
        help()
        exit(RETCODE__NOT_ENOUGHT_ARGUMENTS)

    target_numbers_count = 0
    try:
        target_numbers_count = int(sys.argv[ARGUMENTS_START_SINCE+0])
    except:
        print("ERROR! Cannot parse \"{}\" as integer." . format(sys.argv[ARGUMENTS_START_SINCE+0]))
        help()
        exit(RETCODE__AMOUNT_IS_NOT_INTEGER)

    need_to_print = False
    if len(sys.argv) >= ARGUMENTS_START_SINCE+2:
        if sys.argv[ARGUMENTS_START_SINCE+1].lower() == "print":
            need_to_print = True

    print("Seek {} simple numbers..." . format(target_numbers_count))

    numbers_list = [2,]
    some_number = numbers_list[0]
    begin_time = time.time()
    while len(numbers_list) < target_numbers_count:
      some_number += 1
      simple = is_number_simple(some_number, numbers_list)
      if simple:
        #найдено очередное простое число - добавить его в массив простых чисел
        numbers_list.append(some_number)

    complete_time = time.time();
    duration_seconds = complete_time - begin_time

    if need_to_print:
      print_results(numbers_list)

    msg = "Duration: {:.3f} seconds." . format(duration_seconds)
    msg += " " + "Last simple number = " + str(numbers_list[len(numbers_list)-1])
    print(msg)
